session/session_test.go

```diff
@@ -2415,18 +2415,10 @@ func (s *testSchemaSuite) TestDisableTxnAutoRetry(c *C) {
 	c.Assert(err, NotNil)
 	tk1.MustExec("rollback")
 
-	// other errors could retry
-	config.GetGlobalConfig().TxnLocalLatches.Enabled = true
-	tk1.MustExec("begin")
-	tk2.MustExec("alter table no_retry add index idx(id)")
-	tk2.MustQuery("select * from no_retry").Check(testkit.Rows("8"))
-	tk1.MustExec("update no_retry set id = 10")
-	tk1.MustExec("commit")
-	tk2.MustQuery("select * from no_retry").Check(testkit.Rows("10"))
-
 	// set autocommit to begin and commit
+	config.GetGlobalConfig().TxnLocalLatches.Enabled = true
 	tk1.MustExec("set autocommit = 0")
-	tk1.MustQuery("select * from no_retry").Check(testkit.Rows("10"))
+	tk1.MustQuery("select * from no_retry").Check(testkit.Rows("8"))
```

Comment:
Because I modified the upper case. `error: information schema is changed` could not retry the transaction when `tidb_disable_txn_auto_retry` is on. This only fixes the behavior of `tidb_disable_txn_auto_retry`.
After this PR:
tidb_retry_limit = 0 will disable the transaction retry.
tidb_disable_txn_auto_retry = on will disable the explicit transaction retry.